src/datasets/builder.py

```diff
@@ -286,24 +287,35 @@ def __init__(
             self.info.features = features
 
         # prepare data dirs
-        self._cache_dir_root = os.path.expanduser(cache_dir or config.HF_DATASETS_CACHE)
+        self._cache_dir_root = str(cache_dir or config.HF_DATASETS_CACHE)
```

Comment:
I think we can add a comment here:
```suggestion
        # cache_dir specified by the user can be a remote bucket on GCS or S3
        # This can happen when using Apache Beam with Google Dataflow
        self._cache_dir_root = str(cache_dir or config.HF_DATASETS_CACHE)
```